src/index/options.rs

```diff
@@ -0,0 +1,227 @@
+use std::time::Duration;
+
+use crate::{
+    bson::{serde_helpers, Document},
+    bson_util,
+    collation::Collation,
+};
+
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use typed_builder::TypedBuilder;
+
+/// These are the valid options for specifying an [`IndexModel`](../struct.IndexModel.html).
+/// For more information on these properties, see the [documentation](https://docs.mongodb.com/manual/reference/command/createIndexes/#definition).
+#[serde_with::skip_serializing_none]
+#[derive(Clone, Debug, Default, Deserialize, Serialize, TypedBuilder)]
+#[builder(field_defaults(default, setter(into)))]
+#[serde(rename_all = "camelCase")]
+#[non_exhaustive]
+pub struct IndexOptions {
+    /// Tells the server to build the index in the background and not block other tasks. Starting
+    /// in MongoDB 4.2, this option is deprecated and ignored by the server.
+    pub background: Option<bool>,
+
+    /// Specifies a TTL to control how long MongoDB retains
+    /// documents in this collection.
+    ///
+    /// See the [documentation](https://docs.mongodb.com/manual/core/index-ttl/)
+    /// for more information on how to use this option.
+    #[serde(
+        rename = "expireAfterSeconds",
+        default,
+        deserialize_with = "bson_util::deserialize_duration_option_from_u64_seconds",
+        serialize_with = "bson_util::serialize_duration_option_as_int_secs"
+    )]
+    pub expire_after: Option<Duration>,
+
+    /// Specifies a name outside the default generated name.
+    pub name: Option<String>,
+
+    /// If true, the index only references documents with the specified field. The
+    /// default value is false.
+    ///
+    /// See the [documentation](https://docs.mongodb.com/manual/core/index-sparse/)
+    /// for more information on how to use this option.
+    pub sparse: Option<bool>,
+
+    /// Allows users to configure the storage engine on a per-index basis when creating
+    /// an index.
+    pub storage_engine: Option<Document>,
+
+    /// Forces the index to be unique. The default value is false.
```

Comment:
Can we update this to better describe what "unique" means in this context? Specifically that the collection will not accept duplicate index keys